src/common/docker-compose/index.ts

```diff
@@ -331,8 +331,7 @@ export class DockerComposeUtils {
       if (service.build) {
         if (!service.image) {
           // eslint-disable-next-line unicorn/consistent-destructuring
-          const image = options.getImage ? options.getImage(node.config.metadata.ref) : node.ref;
-          service.image = image;
+          service.image = options.getImage ? options.getImage(node.config.metadata.architect_ref) : node.ref;
```

Comment:
To clarify does the `config.metadata.ref` include the reserved name?

I probably added `architect_ref`, but was immediately confused by this logic.
I'll merge this, but might be worth adding a tech debt ticket since there are 3 refs right now.
`node.ref` // Computed ref - reserved_name
`node.config.metadata.ref` // Feel like this should be the internal ref and remove architect_ref
`node.config.metadata.architect_ref`

Reply:
Registering the service `stateful-api` with the reserved name `stateful-reserved-name`:

* `node.ref` = `stateful-reserved-name`
* `node.config.metadata.ref` = `stateful-reserved-name`
* `node.config.metadata.architect_ref` = `superset.services.stateful-api`

Where by comparison, `stateless-app` that doesn't have a reserved name:

* `node.ref` = `superset--stateless-app`
* `node.config.metadata.ref` = `superset.services.stateless-app`
* `node.config.metadata.architect_ref` = `superset.services.stateless-app`

Reply:
Agreed that that's confusing and I'll make a ticket